include/sentry.h

```diff
@@ -370,6 +370,7 @@ SENTRY_API char *sentry_value_to_json(sentry_value_t value);
  * Sentry levels for events and breadcrumbs.
  */
 typedef enum sentry_level_e {
+    SENTRY_LEVEL_TRACE = -2,
```

Comment:
Even though the level is `-2`, when the logger evaluates whether to return early from `sentry__logger_log` in [this snippet](https://github.com/getsentry/sentry-native/blob/master/src/sentry_logger.c#L93-L96), when the log level is set to debug (e.g. with `sentry_options_set_debug()`) we will still output `SENTRY_TRACE(...)` calls.

```c
    if (g_logger.logger_level != SENTRY_LEVEL_DEBUG
        && level < g_logger.logger_level) {
        return;
    }
```

We could update it so the logger defaults to `sentry_logger_t logger = { NULL, NULL, SENTRY_LEVEL_TRACE };` as the 'lowest level' instead, or alternatively we only output trace logger calls when `sentry_options_set_logger_level(options, SENTRY_LEVEL_TRACE);` was explicitly set (even if running with debug?)

Reply:
I think I added the first clause in the if-statement as a `if debug, log everything` (even though before adding the `trace` level, this would result in the same behavior as not having that check). Now I think just removing the first clause, and only checking whether the level is lower than the set logger_level makes sense.